assets/global.js

```diff
@@ -533,10 +535,17 @@ class VariantSelects extends HTMLElement {
     const newMedia = document.querySelector(
       `[data-media-id="${this.dataset.section}-${this.currentVariant.featured_media.id}"]`
     );
+
     if (!newMedia) return;
+    const modalContent = document.querySelector(`#ProductModal-${this.dataset.section}`);
+    const newMediaModal = modalContent.querySelector( `[data-media-id="${this.currentVariant.featured_media.id}"]`);
     const parent = newMedia.parentElement;
+    if (parent.firstChild == newMedia) return;
+    modalContent.prepend(newMediaModal);
```

Comment:
I just realized that this `modalContent` element is not the right one.

`#ProductModal-${this.dataset.section}` points to the external modal, not its content wrapper.

![image](https://user-images.githubusercontent.com/37168033/130093531-b6692458-faa8-4fad-ba16-8ed87b0ab65a.png)

This is causing the element to be prepended to the wrong container:
![image](https://user-images.githubusercontent.com/37168033/130093648-709b1fc4-82d8-41e2-be28-c8731a6921d4.png)

![image](https://user-images.githubusercontent.com/37168033/130094054-60e8e770-b28d-436c-94e6-e64b0fa1392f.png)

Reply:
🎯